lib/validator/v3/validator.go

```diff
@@ -427,6 +429,49 @@ func validateCIDR(v *validator.Validate, topStruct reflect.Value, currentStructO
 	return err == nil
 }
 
+// validateHTTPMethods validate if the HTTP Method match clauses are valid.
+func validateHTTPMethods(v *validator.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	validMethods := []string{"GET", "PUT", "POST", "HEAD", "DELETE", "OPTIONS", "CONNECT", "*"}
```

Comment:
We have a set library, it might give a cleaner way to express this.
```
var validMethods = set.From("GET", "PUT", "POST", "HEAD", "DELETE", "OPTIONS", "CONNECT", "*")

func validateHTTPMethods(v *validator.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
    ...
    for _, method := range methods {
        valid := validMethods.Contains(method)
        if !valid { ... }
    }
    ....
}
```

Reply:
k. Will remove this restriction.